pkg/s3select/helpers.go

```diff
@@ -0,0 +1,381 @@
+		case *sqlparser.ComparisonExpr:
+			leftVal, _ = matchesMyWhereClause(row, columnNames, alias, left)
+
+		}
+		switch right := expr.Right.(type) {
+		case *sqlparser.ComparisonExpr:
+			rightVal, _ = matchesMyWhereClause(row, columnNames, alias, right)
+		}
+		return (rightVal || leftVal), nil
+
+	}
+	return true, nil
+}
+
+// This is a really important function it actually evaluates the boolean
+// statement and therefore actually returns a bool, it functions as the lowest
+// level of the state machine.
+func evaluateOperator(myTblVal string, operator string, operand interface{}) (bool, error) {
+	if err := checkValidOperator(operator); err != nil {
+		return false, err
+	}
+	myRecordVal := checkStringType(myTblVal)
+
+	myVal := reflect.ValueOf(myRecordVal)
+	myOp := reflect.ValueOf(operand)
+
+	switch {
+	case myVal.Kind() == reflect.String && myOp.Kind() == reflect.String:
+		return stringEval(myVal.String(), operator, myOp.String())
+	case myVal.Kind() == reflect.Float64 && myOp.Kind() == reflect.Float64:
+		return floatEval(myVal.Float(), operator, myOp.Float())
+	case myVal.Kind() == reflect.Int && myOp.Kind() == reflect.Int:
+		return intEval(myVal.Int(), operator, myOp.Int())
+	case myVal.Kind() == reflect.Int && myOp.Kind() == reflect.String:
+		stringVs := strconv.Itoa(int(myVal.Int()))
+		return stringEval(stringVs, operator, myOp.String())
+	case myVal.Kind() == reflect.Float64 && myOp.Kind() == reflect.String:
+		stringVs := strconv.FormatFloat(myVal.Float(), 'f', 6, 64)
+		return stringEval(stringVs, operator, myOp.String())
+	case myVal.Kind() != myOp.Kind():
+		return false, nil
+	}
+	return false, ErrUnsupportedSyntax
+}
+
+// checkValidOperator ensures that the current operator is supported
+func checkValidOperator(operator string) error {
+	listOfOps := []string{">", "<", "=", "<=", ">=", "!=", "like"}
+	for i := range listOfOps {
+		if operator == listOfOps[i] {
+			return nil
+		}
+	}
+	return ErrParseUnknownOperator
+}
+
+// checkStringType converts the value from the csv to the appropriate one.
+func checkStringType(myTblVal string) interface{} {
+	myInt, isInt := strconv.Atoi(myTblVal)
+	myFloat, isFloat := strconv.ParseFloat(myTblVal, 64)
+	if isInt == nil {
+		return myInt
+	} else if isFloat == nil {
+		return myFloat
+	} else {
+		return myTblVal
+	}
+}
+
+// stringEval is for evaluating the state of string comparison.
+func stringEval(myRecordVal string, operator string, myOperand string) (bool, error) {
+	if operator == ">" {
+		return myRecordVal > myOperand, nil
+	} else if operator == "<" {
+		return myRecordVal < myOperand, nil
+	} else if operator == "=" {
+		return myRecordVal == myOperand, nil
+	} else if operator == "<=" {
+		return myRecordVal <= myOperand, nil
+	} else if operator == ">=" {
+		return myRecordVal >= myOperand, nil
+	} else if operator == "!=" {
+		return myRecordVal != myOperand, nil
+	} else if operator == "like" {
+		return wildcard.Match(likeConvert(myOperand), myRecordVal), nil
+	}
```

Comment:
A switch statement for the above would have less brackets and would be nicer to read. Same for intEval and floatEval.